hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -646,6 +647,80 @@ def should_generate_more():
                 count = 0
                 parameter = GenerationParameters(self.random)
 
+            # A thing that is often useful but rarely happens by accident is
+            # to generate the same value at multiple different points in the
+            # test case.
+            #
+            # Rather than make this the responsibility of individual strategies
+            # we implement a small mutator that just takes parts of the test
+            # case with the same label and tries replacing one of them with a
+            # copy of the other.
```

Comment:
This will sound obvious, but I think this comment should explicitly state that we actually *run* the mutated data immediately after creating it.

That way the reader is more likely to look for (and find) the `cached_test_function` call that appears below.